drivers/gpio/gpio_nrfx.c

```diff
@@ -0,0 +1,455 @@
+					: NRF_GPIO_PIN_INPUT_DISCONNECT;
+
+	for (size_t i = 0; i < affected_pins; i++) {
+		u32_t cur_pin = find_lsb_set(pin) - 1;
+
+		pin &= ~BIT(cur_pin);
+
+		nrf_gpio_cfg(
+			NRF_GPIO_PIN_MAP(get_port_cfg(port)->port_num, cur_pin),
+			dir, input, pull, drive, NRF_GPIO_PIN_NOSENSE);
+
+		WRITE_BIT(data->pin_int_en, cur_pin, flags & GPIO_INT);
+		WRITE_BIT(data->trig_edge, cur_pin, flags & GPIO_INT_EDGE);
+		WRITE_BIT(data->double_edge, cur_pin,
+			  flags & GPIO_INT_DOUBLE_EDGE);
+		WRITE_BIT(data->active_level, cur_pin,
+			  flags & GPIO_INT_ACTIVE_HIGH);
+		WRITE_BIT(data->polarity, cur_pin, flags & GPIO_POL_INV);
+
+		gpiote_pin_int_cfg(port, cur_pin);
+	}
+
+	return 0;
+
+}
+
+static int gpio_nrfx_write(struct device *port, int access_op,
+			   u32_t pin, u32_t value)
+{
+	NRF_GPIO_Type *reg = get_port_cfg(port)->port;
+	struct gpio_nrfx_data *data = get_port_data(port);
+
+	if (access_op == GPIO_ACCESS_BY_PORT) {
+		nrf_gpio_port_out_write(reg, value ^ data->polarity);
+	} else {
+		if ((value > 0) ^ ((BIT(pin) & data->polarity) != 0)) {
+			nrf_gpio_port_out_set(reg, BIT(pin));
+		} else {
+			nrf_gpio_port_out_clear(reg, BIT(pin));
+		}
+	}
+
+	return 0;
+}
+
+static int gpio_nrfx_read(struct device *port, int access_op,
+			  u32_t pin, u32_t *value)
+{
+	NRF_GPIO_Type *reg = get_port_cfg(port)->port;
+	struct gpio_nrfx_data *data = get_port_data(port);
+
+	if (access_op == GPIO_ACCESS_BY_PORT) {
+		*value = nrf_gpio_port_in_read(reg);
+		*value ^= data->polarity;
+	} else {
+		*value = (nrf_gpio_port_in_read(reg) & BIT(pin)) >> pin;
+		*value = (BIT(pin) & data->polarity) ? !(*value) : *value;
+	}
+
+	return 0;
+}
+
+static int gpio_nrfx_manage_callback(struct device *port,
+				     struct gpio_callback *callback,
+				     bool set)
+{
+	_gpio_manage_callback(&get_port_data(port)->callbacks, callback, set);
+	return 0;
+}
+
+static int gpio_nrfx_enable_callback(struct device *port,
+				     int access_op,
+				     u32_t pin)
+{
+	struct gpio_nrfx_data *data = get_port_data(port);
+	u8_t pins_affected = 1;
+	int res = 0;
+
+	if (access_op == GPIO_ACCESS_BY_PORT) {
+		pins_affected = popcount(pin);
+	} else {
+		pin = BIT(pin);
+	}
+
+	for (size_t i = 0; i < pins_affected; i++) {
+		u32_t cur_pin = find_lsb_set(pin) - 1;
+
+		pin &= ~BIT(cur_pin);
+
+		if (!(data->pin_int_en & BIT(cur_pin))) {
+			return -EINVAL;
+		}
+
+		data->int_en |= BIT(cur_pin);
+
+		res = gpiote_pin_int_cfg(port, cur_pin);
+		if (res != 0) {
+			return res;
+		}
+	}
+
+	return res;
+}
+
+static int gpio_nrfx_disable_callback(struct device *port,
+				      int access_op,
+				      u32_t pin)
+{
+	struct gpio_nrfx_data *data = get_port_data(port);
+	u8_t pins_affected = 1;
+	int res = 0;
+
+	if (access_op == GPIO_ACCESS_BY_PORT) {
+		pins_affected = popcount(pin);
+	} else {
+		pin = BIT(pin);
+	}
+
+	for (size_t i = 0; i < pins_affected; i++) {
+		u32_t cur_pin = find_lsb_set(pin) - 1;
+
+		pin &= ~BIT(cur_pin);
+
+		data->int_en &= ~BIT(cur_pin);
```

Comment:
This line is the only difference between `gpio_nrfx_enable_callback` and `gpio_nrfx_disable_callback`. You should consider merging these two functions.